pkg/goldpinger/client.go

```diff
@@ -54,6 +55,67 @@ func CheckNeighboursNeighbours(ctx context.Context) *models.CheckAllResults {
 	return CheckAllPods(ctx, SelectPods())
 }
 
+// CheckCluster does a CheckNeighboursNeighbours and analyses results to produce a binary OK or not OK
+func CheckCluster(ctx context.Context) *models.ClusterHealthResults {
+
+	start := time.Now()
+	response := models.ClusterHealthResults{
+		GeneratedAt: strfmt.DateTime(start),
+		OK:          true,
+	}
+	selectedPods := SelectPods()
+
+	// precompute the expected set of nodes
+	expectedNodes := []string{}
+	for _, peer := range selectedPods {
+		expectedNodes = append(expectedNodes, peer.HostIP)
+	}
+	sort.Strings(expectedNodes)
+
+	// get the response we serve for check_all
+	checkAll := CheckAllPods(ctx, selectedPods)
+
+	// we should at the very least have a response from ourselves
+	if len(checkAll.Responses) < 1 {
+		response.OK = false
+	}
+	for _, resp := range checkAll.Responses {
+		// 1. check that all nodes report OK
+		if *resp.OK {
+			response.NodesHealthy = append(response.NodesHealthy, resp.HostIP.String())
+		} else {
+			response.NodesUnhealthy = append(response.NodesUnhealthy, resp.HostIP.String())
+			response.OK = false
+		}
+		response.NodesTotal++
+		// 2. check that all nodes report the expected peers
+		// on error, there might be no response from the node
+		if resp.Response == nil {
+			response.OK = false
+			break
```

Comment:
If we break here, the list of `response.NodesHealthy` and `response.NodesUnhealthy` will be incomplete. Perhaps a `continue`?